### ISSUE TITLE
[json-tests] Sort out the logging as tests run

### ISSUE THREAD

**Issue by dvdplm** [openethereum/openethereum#11084]
The logging of activity during json-test runs is a bit odd, with copious use of `flushln!` and `flush!` which makes it hard to follow what is printed where.
Switch to using proper logging with `info!`/`warn!`/`debug!`/`trace` as appropriate. Use `target: "json-tests"` to make filtering easy.

**Comment by Ishaan28malik** [comment]
Is it still open and unassigned ?

**Comment by dvdplm** [comment]
Yes! :)

**Comment by dvdplm** [comment]
@Ishaan28malik to run the json-tests, issue a `cargo test --features=json-tests` in the `ethcore/` subfolder. When running all the tests it's often faster overall to use `--release` as well but you'll want to run single tests as well, e.g. `cargo test --features=json-tests stSpecialTest` (this will run all tests in the `res/ethereum/tests/BlockchainTests/GeneralStateTests/stSpecialTest/` folder).

**Comment by dvdplm** [comment]
@Ishaan28malik let us know if we can do anything to assist you here. :)

**Comment by rakanalh** [comment]
@dvdplm I had a go at this... PR mentioned before this comment. 

However, when i run: `cargo test --features=json-tests`

Some of the tests would fail with:
```
---- json_tests::difficulty::difficulty_test_homestead::DifficultyTests_difficultyHomestead stdout ----
thread 'json_tests::difficulty::difficulty_test_homestead::DifficultyTests_difficultyHomestead' panicked at 'Error opening test file at: "res/ethereum/tests/BasicTests/difficultyHomestead.json"', ethcore/src/json_tests/test_common.rs:103:13
```

And `ethcore/res/ethereum/tests` directory is empty. Are those files generated somehow?

**Comment by dvdplm** [comment]
@rakanalh Thank you for stepping up! :)

So the json tests are contained in git submodules, `git submodule update --init --recursive` should set that up for you. Here's what I have on my machine:

```
$ git submodule status
 d4f86ecf4aa7c44a40bc0c972fd3e25d63ef5d92 ethcore/res/ethereum/tests (PoC6-2175-gd4f86ecf4)
 0edbf860ff7ed4b6b6336097ba44836e8c6482dd ethcore/res/wasm-tests (remotes/origin/call-gasleft)
```

Does that match with what's on your machine?

**Comment by rakanalh** [comment]
Ahh, didn't check if the repo had submodules. Yes, the commit hashes match what i have. Thanks alot

**Comment by rakanalh** [comment]
Now getting 
```
ERROR 2019-10-27T15:19:01Z: ethcore::miner::miner: Error occurred while refreshing service transaction cache: please ensure the contract and method you're calling exist! failed to decode empty bytes. if you're using jsonrpc this is likely due to jsonrpc returning `0x` in case contract or method don't exist
```
when i run the tests, i'll check it out.

**Comment by dvdplm** [comment]
@rakanalh Yeah I've seen that too. It is most likely a regression introduced fairly recently and I think it's spurious (in the context of the json tests at least).

/cc @tomusdrw you mentioned you had a hunch on what PR it might be that introduced it?

**Comment by tomusdrw** [comment]
Introduce here https://github.com/paritytech/parity-ethereum/pull/10088 it seems, I thought it's actually more recent, but it seems it went in over 6 months ago.
